packages/p2p/src/components/my-ads/my-ads-row-renderer.jsx

```diff
@@ -63,24 +69,30 @@ const MyAdsRowRenderer = observer(({ row: advert, setAdvert }) => {
                 is_left_swipe
                 right_hidden_component={
                     <React.Fragment>
-                        <div className='p2p-my-ads__table-popovers__edit' onClick={onClickEdit}>
-                            <Icon custom_color='var(--general-main-1)' icon='IcEdit' size={16} />
-                        </div>
-                        <div className='p2p-my-ads__table-popovers__activate'>
-                            <Icon
-                                icon='IcArchive'
-                                custom_color='var(--general-main-1)'
-                                size={14}
-                                onClick={onClickActivateDeactivate}
-                            />
+                        <div className='p2p-my-ads__table-popovers__edit' onClick={() => handleOnEdit()}>
+                            <Icon custom_color='var(--general-main-1)' icon='IcEdit' />
                         </div>
+                        {is_advert_active ? (
+                            <div className='p2p-my-ads__table-popovers__activate'>
+                                <Icon
+                                    icon='IcArchive'
+                                    custom_color='var(--general-main-1)'
+                                    size={14}
+                                    onClick={onClickActivateDeactivate}
+                                />
+                            </div>
+                        ) : (
+                            <div className='p2p-my-ads__table-popovers__deactivate'>
+                                <Icon
+                                    icon='IcUnarchive'
+                                    custom_color='var(--general-main-1)'
+                                    size={14}
+                                    onClick={onClickActivateDeactivate}
+                                />
+                            </div>
+                        )}
```

Comment:
Maybe instead of having the ternary on two different divs, we could have 1 div, and have the ternaries for className, and icon as there would be less lines of code and a bit more readable: 

```
<div className={`p2p-my-ads__table-popovers__${is_advert_active ? 'activate' : 'deactivate'}`}>
        <Icon
            icon={`${is_advert_active ? 'IcArchive' : 'IcUnarchive'}`}
            custom_color='var(--general-main-1)'
            size={14}
            onClick={onClickActivateDeactivate}
        />
</div>
```

Reply:
Will refactor the code